pulser-core/pulser/backend/noise_model.py

```diff
@@ -111,6 +111,16 @@ def __post_init__(self) -> None:
                     DeprecationWarning,
                 )
                 if prob != rate:
+                    if (
+                        len(rate) > 0
+                        if noise == "eff_noise"
+                        else rate
+                        != self.__dataclass_fields__[rate_name].default
```

Comment:
I know it's more verbose but I think it would be preferable to search through `dataclasses.fields(self)` to get the `rate` field and then get the default value from it.

There's no guarantee that the structure of the internal won't change, so we shouldn't access it.